zerver/worker/queue_processors.py

```diff
@@ -609,3 +611,33 @@ def consume(self, event: Mapping[str, Any]) -> None:
                 (stream, recipient, sub) = access_stream_by_id(user_profile, stream_id,
                                                                require_active=False)
                 do_mark_stream_messages_as_read(user_profile, client, stream)
+        elif event['type'] == 'realm_exported':
+            realm = get_realm(event['realm_name'])
+            output_dir = tempfile.mkdtemp(prefix="zulip-export-")
+
+            to_s3 = False
+            if settings.LOCAL_UPLOADS_DIR is None:
+                to_s3 = True
+
+            export_realm_wrapper(realm=realm, output_dir=output_dir,
+                                 upload_to_s3=to_s3, threads=6, public_only=True,
+                                 delete_after_upload=to_s3)
+
+            admins = [user for user in realm.get_admin_users()]
+            content = "A public-only export was successfully completed!\n"
+            if to_s3:
+                content += (
+                    "The data was uploaded to the following url:\n\n"
+                    "http://%s.amazonaws.com/%s/%s" %
+                    (settings.S3_REGION, settings.S3_AUTH_UPLOADS_BUCKET, settings.S3_KEY,)
```

Comment:
This line is broken, bad style, and a serious security issue!  As I mention below, it's bad style because it duplicate a computation already happening in `export_realm_wrapper`.  It's broken because the URL you construct here doesn't match what `export_realm_wrapper` does, and a serious security issue because you actually include the secret `S3_KEY` in your incorrect URL.